x-pack/legacy/plugins/lens/public/editor_frame_plugin/editor_frame/suggestion_helpers.ts

```diff
@@ -26,13 +30,58 @@ export interface Suggestion {
  * Each suggestion represents a valid state of the editor and can be applied by creating an
  * action with `toSwitchAction` and dispatching it
  */
-export function getSuggestions(
-  datasourceTableSuggestions: DatasourceSuggestion[],
-  visualizationMap: Record<string, Visualization>,
-  activeVisualizationId: string | null,
-  visualizationState: unknown
-): Suggestion[] {
-  const datasourceTables: TableSuggestion[] = datasourceTableSuggestions.map(({ table }) => table);
+export function getSuggestions({
+  datasourceMap,
+  datasourceStates,
+  visualizationMap,
+  activeVisualizationId,
+  visualizationState,
+  field,
+}: {
+  datasourceMap: Record<string, Datasource>;
+  datasourceStates: Record<
+    string,
+    {
+      isLoading: boolean;
+      state: unknown;
+    }
+  >;
+  visualizationMap: Record<string, Visualization>;
+  activeVisualizationId: string | null;
+  visualizationState: unknown;
+  field?: unknown;
+}): Suggestion[] {
+  const allLayerIds = _.flatten(
+    Object.entries(datasourceMap).map(([datasourceId, datasource]) => {
```

Comment:
You are right, it makes things easier. However, I would leave the refactoring to a separate PR as it would touch a lot of places and this one is already way too big. The duplicate filter is a good point, I unified that.